File: Abstract-Factory-Pattern/src/com/sosa/factories/ShapeFactory.java
package com.sosa.factories;
import com.sosa.infrastructure.*;

public class ShapeFactory extends AbstractFactory {

    private Shape shape = null;

    @Override
    public Shape getShape(ShapeTypes type) {
        switch (type){
            case CIRCLE: {
                shape = new Circle();
                break;
            }
            case RECTANGLE: {
                shape = new Rectangle();
                break;
            }
            case SQUARE: {
                shape = new Square();
                break;
            }
            default: {
                throw new UnsupportedOperationException();
            }
        }
        return shape;
    }
}

File: Factory-Pattern/Infrastructure/ShapeFactory.cs
namespace Factory_Pattern.Infrastructure.Shapes {
    public class ShapeFactory {
        private IShape _shape = null;

        public IShape createShape (Shapes shapeType) {
            switch (shapeType) {
                case Shapes.Circle:
                    {
                        _shape = new Circle ();
                        break;
                    }
                case Shapes.Rectangle:
                    {
                        _shape = new Rectangle ();
                        break;
                    }
                case Shapes.Square:
                    {
                        _shape = new Square ();
                        break;
                    }

                default:
                    {
                        throw new System.NotImplementedException ();
                    }

            }

            return _shape;
        }
    }
}